src/screens/Users/AddOrEditUser.js

```diff
@@ -144,48 +472,102 @@ class AddOrEditUser extends React.PureComponent<Props, State> {
     this.setState({ visibleModal: '' });
   };
 
-  verifyEmail = () => {
-    this.setState({ verifyingField: 'email' });
-  };
-
-  verifyPhone = () => {
-    this.setState({ verifyingField: 'phone' });
+  verifyField = (verifyingField) => {
+    this.setState({ verifyingField });
   };
 
   onCloseVerification = () => {
-    this.setState({ verifyingField: null });
+    const { verifyingField } = this.state;
+    this.setState({ verifyingField: null, verifiedModalField: verifyingField });
```

Comment:
No matter if I haven't verified my data and I close the modal by clicking on "x" icon, VerifiedModal is still being shown to me saying that my data is verified.

Reply:
good catch :)